src/util/i18n.js

```diff
@@ -1,18 +1,52 @@
+import { DEFAULT_LANGUAGE, I18N_DICTIONARY } from './PinConst';
+
+class I18n {
+
+    /**
+     * @property
+     * @type {string} The currently set language
+     */
+    get lang() {
+        return this._lang;
+    }
 
-/**
- * This is a wrapper for all things related to internationalization
- */
-export default {
-    lang: 'en',
     /**
      * If the lang is supported (found has a key in Strings), set new lang.
+     * @property
      * @param {string} _lang - the language code to set
      */
-    setLang: function(lang) {
-        if (Strings[lang]) {
-            this.lang = lang;
+    set lang(lang) {
+        if (this.dictionary.hasOwnProperty(lang)) {
+            this._lang = lang;
         }
-    },
+    }
+
+    /**
+     * @property
+     * @type {Object} map of languages and strings
+     */
+    get dictionary() {
+        return this._dictionary;
+    }
+
+    /**
+     * @property
+     * @param  {object} dictionary
+     */
+    set dictionary(dictionary) {
+        this._dictionary = dictionary;
+    }
+
+    /**
+     * Create a new i18n class to translate strings
+     * @param {Object} dictionary
+     * @param {string} language Initial language for the dictionary
+     */
+    constructor(dictionary, language) {
+        this._lang = language || DEFAULT_LANGUAGE;
```

Comment:
Don't need the `|| DEFAULT_LANGUAGE` since that's exactly what you're passing in, and it's a singleton.